### PR TITLE
[1250] support cache control with .env

### LINKED ISSUE #1250: [Feature] Configurable Cache Control Header
### Description of Feature

For the `/api/features/*` endpoints, we hard-code a Cache-Control header in the response:

```
public, max-age=30, stale-while-revalidate=3600, stale-if-error=36000
```

This is optimized for GrowthBook Cloud's CDN use case.  We should make this configurable via environment variables for those who are self-hosting and want more control over caching.

### PR DESCRIPTION
**Pull request by gazzdingo**
… your env

### Features and Changes
adding support so that you can add your own cache controls in .env 
supported functions are:
`max-age`
`stale-while-revalidate`
`stale-if-error`

if you want to add support you need to add here is an example that you can add to your .env file
```
CACHE_CONTROL_MAX_AGE=10
CACHE_CONTROL_STALE_WHILE_REVALIDATE=100
CACHE_CONTROL_STALE_IF_ERROR=1000
```

- Closes #1250

### Testing

- in you env.local add example code above
- create a new sdk feature 
image
- copy the url that is generated eg `http://localhost:3100/api/features/sdk-blablaba`
- inspect and check the headers and make sure the cache control is correct 
image

### CODE REVIEW

File: packages/back-end/src/controllers/features.ts [hunk 1]
```diff
@@ -190,10 +195,10 @@ export async function getFeaturesPublic(req: Request, res: Response) {
       hashSecureAttributes,
     });
 
-    // Cache for 30 seconds, serve stale up to 1 hour (10 hours if origin is down)
+    // The default is Cache for 30 seconds, serve stale up to 1 hour (10 hours if origin is down)
     res.set(
       "Cache-control",
-      "public, max-age=30, stale-while-revalidate=3600, stale-if-error=36000"
+      `public, max-age=${CACHE_CONTROL_MAX_AGE}, stale-while-revalidate=${CACHE_CONTROL_STALE_IF_ERROR}, stale-if-error=${CACHE_CONTROL_STALE_WHILE_REVALIDATE}`
```
Comment (jdorn) [inline packages/back-end/src/controllers/features.ts:201]: `stale-if-error` and `stale-while-revalidate` look swapped here

File: packages/back-end/src/util/secrets.ts [hunk 2]
```diff
@@ -141,6 +141,16 @@ export const QUERY_CACHE_TTL_MINS =
 export const IMPORT_LIMIT_DAYS =
   parseInt(process.env?.IMPORT_LIMIT_DAYS || "") || 365;
 
+// cache control currently feature only /api/features/*
+export const CACHE_CONTROL_MAX_AGE =
+  parseInt(process.env?.CACHE_CONTROL_MAX_AGE || "") || 30;
+export const CACHE_CONTROL_STALE_WHILE_REVALIDATE =
+  parseInt(process.env?.CACHE_CONTROL_MAX_AGE || "") || 3600;
+export const CACHE_CONTROL_STALE_IF_ERROR =
+  parseInt(process.env?.CACHE_CONTROL_MAX_AGE || "") || 36000;
```
Comment (jdorn) [inline packages/back-end/src/util/secrets.ts:150]: Looks like a copy/paste error here

**Review by jdorn — CHANGES_REQUESTED** [review]

File: packages/back-end/src/controllers/features.ts (hunk 1 above)
Comment (gazzdingo) [reply packages/back-end/src/controllers/features.ts:201]: cood catch

File: packages/back-end/src/util/secrets.ts (hunk 2 above)
Comment (gazzdingo) [reply packages/back-end/src/util/secrets.ts:150]: yep it was forgot to update the value 👍

**Review by bryce-fitzsimons — APPROVED** [review]
lgtm

**Review by jdorn — APPROVED** [review]

**Pull request merged**